packages/components/organisms/f-checkout/src/components/Checkout.vue

```diff
@@ -335,23 +324,29 @@ export default {
             return !this.hasCheckoutLoadedSuccessfully;
         },
 
-        shouldShowErrorDialog () {
-            return this.nonFulfillableError ? this.nonFulfillableError.shouldShowInDialog : false;
-        },
-
-        restaurantMenuPageUrl () {
-            return `restaurant-${this.restaurant.seoName}/menu`;
-        },
-
-        hasNonFulfillableErrorCode () {
-            return this.nonFulfillableError && this.nonFulfillableError.code;
-        },
-
         eventData () {
             return {
                 isLoggedIn: this.isLoggedIn,
                 serviceType: this.serviceType
             };
+        },
+
+        messageType () {
+            const dialog = {
+                name: 'error-dialog'
+            };
+
+            const alert = {
+                name: 'alert',
+                props: {
+                    type: 'danger',
+                    class: this.$style['c-checkout-alert'],
+                    heading: this.$t('errorMessages.errorHeading')
+                },
+                content: this.message
+            };
```

Comment:
I'd probably extract these two into two separate computed properties.

So you'd end up with something like this:

```
messageType () {
  return this.message && !this.message.shouldShowInDialog
    ? this.alertMessage
    : this.dialogMessage;
}
```